internal/pkg/client/patch.go

```diff
@@ -0,0 +1,207 @@
+			if mergepatch.IsPreconditionFailed(err) {
+				return patch, fmt.Errorf("at least one of apiVersion, kind and name was changed for %s", resourceDescription)
+			}
+			return patch, err
+		}
+	case err != nil:
+		return patch, err
+	case err == nil:
+		// Compute a three way strategic merge patch to send to server.
+		patch.Type = types.StrategicMergePatchType
+		lookupPatchMeta, err := strategicpatch.NewPatchMetaFromStruct(versionedObject)
+		if err != nil {
+			return patch, err
+		}
+		patch.Data, err = strategicpatch.CreateThreeWayMergePatch(last, modified, currentb, lookupPatchMeta, true)
+		if err != nil {
+			return patch, err
+		}
+	}
+	return patch, nil
+}
+
+// GetMergePatch - generate merge patch from original and modified objects
+func GetMergePatch(original, modified runtime.Object) (Patch, error) {
+	originalJSON, err := json.Marshal(original)
+	if err != nil {
+		return Patch{}, err
+	}
+
+	modifiedJSON, err := json.Marshal(modified)
+	if err != nil {
+		return Patch{}, err
+	}
+
+	patch, err := jsonpatch.CreateMergePatch(originalJSON, modifiedJSON)
+	if err != nil {
+		return Patch{}, err
+	}
+	return Patch{types.MergePatchType, patch}, nil
+}
+
+// GetServerSidePatch - generate merge patch from original and modified objects
+func GetServerSidePatch(original, modified runtime.Object) (Patch, error) {
+	return Patch{}, nil
+}
+
+// getLastApplied retrieves the original configuration of the object
+// from the annotation, or nil if no annotation was found.
+func getLastApplied(obj runtime.Object) ([]byte, error) {
+	annots, err := metadataAccessor.Annotations(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	if annots == nil {
+		return nil, nil
+	}
+
+	original, ok := annots[v1.LastAppliedConfigAnnotation]
+	if !ok {
+		return nil, nil
+	}
+
+	return []byte(original), nil
+}
+
+// GetModifiedConfiguration retrieves the modified configuration of the object.
+// If annotate is true, it embeds the result as an annotation in the modified
+// configuration. If an object was read from the command input, it will use that
+// version of the object. Otherwise, it will use the version from the server.
+func GetModifiedConfiguration(obj runtime.Object, annotate bool, codec runtime.Encoder) ([]byte, error) {
+	// First serialize the object without the annotation to prevent recursion,
+	// then add that serialization to it as the annotation and serialize it again.
+	var modified []byte
+
+	// Otherwise, use the server side version of the object.
+	// Get the current annotations from the object.
+	annots, err := metadataAccessor.Annotations(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	if annots == nil {
+		annots = map[string]string{}
+	}
+
+	original := annots[v1.LastAppliedConfigAnnotation]
+	delete(annots, v1.LastAppliedConfigAnnotation)
+	if err := metadataAccessor.SetAnnotations(obj, annots); err != nil {
+		return nil, err
+	}
+
+	modified, err = runtime.Encode(codec, obj)
+	if err != nil {
+		return nil, err
+	}
+
+	if annotate {
```

Comment:
don't put this conditional in if it is never false

Reply:
it is false in the test code.